### PR TITLE
Add the PyGMT ecosystem page

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Address #3371.

Adapted from https://geopandas.org/en/latest/community/ecosystem.html.

**Preview**: https://pygmt-dev--3475.org.readthedocs.build/en/3475/ecosystem.html

### CODE REVIEW

File: doc/ecosystem.md
Comment (michaelgrund) [inline doc/ecosystem.md:10]:
```suggestion
In addition to these core dependencies, PyGMT also relies on several optional packages to
```

File: doc/ecosystem.md
Comment (michaelgrund) [inline doc/ecosystem.md:51]:
```suggestion
PyGMT relies on IPython to provide a rich interactive experience in Jupyter notebooks.
```

File: doc/ecosystem.md
Comment (michaelgrund) [inline doc/ecosystem.md:81]:
```suggestion
Currently, PyGMT relies on [rioxarray][] to save multi-band rasters to temporary files
```

**Comment by michaelgrund** [comment]
Do you plan to also add the packages that rely on PyGMT as mentioned in the bottom of the page in this PR?

**Comment by seisman** [comment]
> Do you plan to also add the packages that rely on PyGMT as mentioned in the bottom of the page in this PR?

I feel we shouldn't add any packages without asking for permission from the package maintainers. 

Similar to what geopandas is doing (https://geopandas.org/en/stable/community/ecosystem.html), we may add a note like:

> If your package relies on PyGMT, please let us know or add it by yourself.

File: doc/ecosystem.md
Comment (michaelgrund) [reply doc/ecosystem.md:30]: Actually it's *pandas* instead of *Pandas*.

Comment (yvonnefroehlich) [reply doc/ecosystem.md:30]: True, just checked the docs.
Hm, then leave the text as is. But maybe keep upper case for the heading, as it also looks a bit strange when we mix upper and lower case?

Comment (seisman) [reply doc/ecosystem.md:30]: The pandas documentation uses `pandas` everywhere (e.g., https://pandas.pydata.org/docs/getting_started/install.html). I think `pandas` is just like a trademark and should not be changed to `Pandas` even at the beginning of a sentence.

File: doc/ecosystem.md
Comment (seisman) [reply doc/ecosystem.md]: I've changed it to "rioxarray" to match the upstream documentation.

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:17]: I'm thinking if we should remove the headings "Required dependencies"/"Optional dependencies" and list all packages under the "PyGMT dependencies" heading. Required dependencies are marked by an asterisk (e.g., `### NumPy*`).

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:21]: The NumPy description comes from https://numpy.org/doc/stable/user/whatisnumpy.html#whatisnumpy.

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:30]: This paragraph comes from https://pandas.pydata.org/docs/getting_started/overview.html.

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:35]: Maybe remove the last two sentences because it's not directly related to Python.
```suggestion
data analysis in Python.
```

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:39]: This paragraph comes from https://xarray.dev/.

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:41]: Required packages "packaging" and "netCDF4" are not documented yet. Do we want to document these two packages?

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:47]: This paragraph comes from https://ipython.readthedocs.io/en/stable/#ipython-documentation

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:55]: xref: https://geopandas.org/en/stable/

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:66]: xref: https://contextily.readthedocs.io/en/latest/

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:69]: Remove the last sentence because it's about the details.
```suggestion
geospatial raster files.
```

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md]: I can't remember where I found the rioxarray description. Maybe it's written by ChatGPT.

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:92]: xref: https://arrow.apache.org/docs/index.html and https://arrow.apache.org/docs/python/.

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:99]: This note is copied from https://www.pygmt.org/dev/install.html with updates from PR #2933. 

The plan is to remove the note from https://www.pygmt.org/dev/install.html. See PR #3506.

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:112]: For a package that rely on PyGMT, I think a short description about the package itself and how PyGMT is used in the package is enough. 

For example, I know [SHTOOLS](https://github.com/SHTOOLS/SHTOOLS) maintained by `MarkWieczorek` uses PyGMT. I expect to see it listed here like below:
```
### SHTOOLS

pyshtools is an archive of Python software that can be used for spherical harmonic transforms, multitaper spectral analyses, expansions of gridded data into Slepian basis functions, and standard operations on global gravitational and magnetic field data.

It uses PyGMT for the creation of publication quality maps.
```

I don't plan to add SHTOOLS myself. Instead, after merging this PR, I prefer to ping `MarkWieczorek` to see if he is willing to add SHTOOLS himself.

**Comment by seisman** [comment]
@GenericMappingTools/pygmt-maintainers I think this PR is ready for final reviews.

File: doc/ecosystem.md
Comment (seisman) [inline doc/ecosystem.md:98]:
```suggestion
Specifically, only uint/int/float and date32/date64 are supported for now.
Support for string Array dtypes, Duration types and GeoArrow geometry types is still a work in progress.
```

File: doc/ecosystem.md
Comment (seisman) [reply doc/ecosystem.md:94]: It's likely this PR will be merged before PR #2933, so I've reverted the changes in PR #2933 (770968e96593735890532fc3de09931b5b78ecef) and use the original notes from the installation guides.

File: doc/index.md
Comment (yvonnefroehlich) [inline doc/index.md:58]: I am wondering if it makes sense to have `ecosystem.md` in one section together with `minversions.md`. Currently `minversions.md` is listed under "Reference documentation" and `ecosystem.md` under "Development".

Comment (seisman) [reply doc/index.md]: Makes sense. Done in 5bf8c5dbeeb62cf6e207a7743bfdbe19bef2628d.

File: doc/ecosystem.md
Comment (yvonnefroehlich) [reply doc/ecosystem.md:107]: Sounds good to me.

**Comment by seisman** [comment]
Ping @GenericMappingTools/pygmt-maintainers for final reviews. Will merge in 48 hours if no further comments.

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**
